components/cloud/azure/src/azblob.rs

```diff
@@ -0,0 +1,837 @@
+                "account name cannot be empty to access azure blob storage",
+            ))
+        }
+    }
+
+    pub fn parse_plaintext_account_url(&self) -> Option<String> {
+        if let (Some(account_name), Some(shared_key)) =
+            (self.account_name.as_ref(), self.shared_key.as_ref())
+        {
+            let blob_endpoint = match self.bucket.endpoint.as_ref() {
+                Some(s) => s.to_string(),
+                None => format!("https://{}.blob.core.windows.net", account_name),
+            };
+            Some(
+                ConnectionStringBuilder::new()
+                    .account_name(account_name)
+                    .account_key(shared_key)
+                    .blob_endpoint(&blob_endpoint)
+                    .build(),
+            )
+        } else {
+            None
+        }
+    }
+
+    pub fn parse_env_plaintext_account_url(&self) -> Option<String> {
+        if let (Some(account_name), Some(shared_key)) =
+            (self.env_account_name.as_ref(), self.env_shared_key.as_ref())
+        {
+            let blob_endpoint = match self.bucket.endpoint.as_ref() {
+                Some(s) => s.to_string(),
+                None => format!("https://{}.blob.core.windows.net", account_name),
+            };
+            Some(
+                ConnectionStringBuilder::new()
+                    .account_name(account_name)
+                    .account_key(shared_key)
+                    .blob_endpoint(&blob_endpoint)
+                    .build(),
+            )
+        } else {
+            None
+        }
+    }
+}
+
+impl BlobConfig for Config {
+    fn name(&self) -> &'static str {
+        STORAGE_NAME
+    }
+
+    fn url(&self) -> io::Result<url::Url> {
+        self.bucket.url("azure").map_err(|s| {
+            io::Error::new(
+                io::ErrorKind::Other,
+                format!("error creating bucket url: {}", s),
+            )
+        })
+    }
+}
+
+enum RequestError {
+    IO(io::Error),
+    InvalidInput(Box<dyn std::error::Error + Send + Sync>, String),
+    TimeOut(String),
+}
+
+impl From<RequestError> for io::Error {
+    fn from(err: RequestError) -> Self {
+        match err {
+            RequestError::IO(e) => e,
+            RequestError::InvalidInput(e, tag) => {
+                Self::new(io::ErrorKind::InvalidInput, format!("{}: {}", tag, &e))
+            }
+            RequestError::TimeOut(msg) => Self::new(io::ErrorKind::TimedOut, msg),
+        }
+    }
+}
+
+impl RetryError for RequestError {
+    fn is_retryable(&self) -> bool {
+        matches!(self, Self::TimeOut(_))
+    }
+}
+
+/// Specifies the minimum size to use multi-block upload.
+/// Azure requires each part to be at least 4 MiB compatible with 2016 version.
+/// https://docs.microsoft.com/en-us/rest/api/storageservices/understanding-block-blobs--append-blobs--and-page-blobs
+/// while azure-sdk-for-go use 256 MiB
+const MINIMUM_BLOCK_SIZE: usize = 128 * 1024 * 1024; // 128MB
```

Comment:
When a file reaches 96M, we will split it into multi files. The multi-part upload may never be triggered if we set the block size to 128M.

Reply:
Done. Please take another look.